### ISSUE TITLE
'module name' and submodule

### ISSUE THREAD

**Issue by stupidisum** [michaelkrone/generator-material-app#45]
Do you have time to do this?
Maybe our group can join you maintaining this repo.

**Comment by michaelkrone** [comment]
Hi @stupidisum,

what do you mean with 'module name'?
I would be glad if you would like to help maintaining this repository.

**Comment by stupidisum** [comment]
I found get complex route done without some trifle with module name.
I think it will be great if apiroute can do this:
```
yo material-app:apiroute admin
yo material-app:apiroute admin.user
yo material-app:apiroute admin.main
yo material-app:apiroute admin.user.edit.password
```

I did something like this before in another [generator](https://github.com/stupidisum/generator-ng-zhaimi/blob/master/route/index.js#L19-L28)

**Comment by michaelkrone** [comment]
Yes, this would be great.
What would you do if the 'parent' module does not exist yet?

**Comment by stupidisum** [comment]
I just asked if user needs an abstract father
https://github.com/stupidisum/generator-ng-zhaimi/blob/master/route/index.js#L52-L69

**Comment by stupidisum** [comment]
But my work with templates was bad. I just created the abstract route in the same file.
https://github.com/stupidisum/generator-ng-zhaimi/blob/master/templates%2Froute%2Froute(uirouter).js#L27-L42

**Comment by michaelkrone** [comment]
Looks good, an optional abstract parent seems a good approach to me.
Would you like to dig into this? I could investigate in this feature too, but I am preparing an Angular Material update, so it would took some time.

**Comment by stupidisum** [comment]
I think I will do this in this week if it's not too complex. 
Our project needs this feature, but it needs time, too.

Is there any way to let apiroute generate the father routes one by one?
Or just add an abstract route like my old work do.

**Comment by michaelkrone** [comment]
There is the possibility to use [generator.composeWith](http://yeoman.io/authoring/composability.html) and call the subgenerators. Or to call [createFiles and registerRoute](https://github.com/michaelkrone/generator-material-app/blob/master/generators/apiroute/index.js#L86) of the apiroute generator by hand. I think composeWith is a cleaner approach, what do you think?

**Comment by stupidisum** [comment]
Yes, it's much better. Just like the `mkdir` command.

So, if I already have routes `a` and `a.b` and I want to generate a new route `a.b.c.d.e`
Command `material-app:apiroute a.b.c.d.e` is all I need and it's the same as the following commands. Module name should be exactly same as route name by default.

```
material-app:apiroute a.b.c
material-app:apiroute a.b.c.d
material-app:apiroute a.b.c.d.e
```

Here are some problems needs to be figured out.
1. How can generator tell `a.b` already exists?
2. Should the generator add new child-modules' names into father routes and how?

**Comment by michaelkrone** [comment]
1. The generated .yo-rc.json contains an angularModules section, which contains modules that have been registered by the generator. Since a route is wrapped in a module, we could lookup registered modules here or add a section for registered routes.

2. That's a harder one, we could adjust the template to insert child routes on some conditions, I do not know right now, have to think about it.

**Comment by stupidisum** [comment]
Hey, buddy. Do you have time to maintain this repo any more?
How about giving our group permissions to maintain this repo?

**Comment by michaelkrone** [comment]
Hi @stupidisum,

at the moment, I do not have much time for this project. What is you group about? I would like to grant you access for developing.

**Comment by Stupidism** [comment]
We are IT department of [ClarityMovement](https://github.com/ClarityMovement)

https://clarity.io/

**Comment by Stupidism** [comment]
Sorry, I forgot to label those PRs as WIP. After you merged them, I commited something else. 
So I decided to `reset` back to 0.1.5 and merge master branch of my fork. 
Sorry again for `push --force` in master.

Please add a protection for master branch in settings. I can't change the settings of this repo.
And please set develop as default branch!
From now, I will develop in git-flow-mode

**Comment by michaelkrone** [comment]
Done. I'll close this Issue now.

Closed with #48
